src/packet_analysis/Analyzer.cc

```diff
@@ -166,30 +166,64 @@ void Analyzer::Weird(const char* name, Packet* packet, const char* addl) const
 	session_mgr->Weird(name, packet, addl, GetAnalyzerName());
 	}
 
+void Analyzer::EnqueueAnalyzerConfirmationInfo(session::Session* session, zeek::Tag arg_tag)
```

Comment:
Where is the check if we have session at all, and only generate an event if we do? Is that up to the caller to check? I'd add a check into these methods if session is null, easy to break otherwise.

Same for the methods below.

Reply:
I was forth and back on that and minimally the crash indicated that one is doing something wrong. Calling AnalyzerConfirmation/Violation for packet analyzers without sessions currently isn't valid. Protecting from it makes things safer, but also swallows the confirmation/violation that someone may expect.

I'll add it, but if we have some sort of WarnOnce() I'd probably like to use that here.